docs/multitenancy/proposal.md

```diff
@@ -0,0 +1,81 @@
+
+### CRDs and webhooks
+
+`CustomResourceDefinitions`, `ValidatingWebhookConfigurations` and `MutatingWebhookConfigurations` are cluster-wide so they can't be managed on a per-operator basis.
+In a multitenant configuration installing and managing the CRDs and webhooks will be a separate step from deploying operators (and will require cluster admin privileges).
+
+The conversion webhooks in the CRDs and the validating and mutating webhook configurations that are defined for different ASO resources will all point to the `azureoperator-webhook-service` service deployed in the `azureoperator-system` namespace (by default).
+The service will be backed by a deployment of the normal ASO operator running in a restricted webhook-only mode - it won't be watching any namespaces for Azure resources and won't have access to any Azure credentials.
+This mode will be triggered by the configuration setting `AZURE_WEBHOOK_ONLY=1`.
+With this setting no other configuration values will be required, and no resource watches will be started - the operator will just respond to webhook requests from the API server.
+
+### RBAC
+
+At the moment installing a global instance of ASO creates a `ClusterRole` defining all of the permissions the operator requires on the different Azure resources (`azureoperator-manager-role`), and then a `ClusterRoleBinding` granting those permissions for resources in all namespaces (`azureoperator-manager-rolebinding`).
+A multitenant deployment won't have the global `ClusterRoleBinding`; instead each managed namespace N will need to contain a `RoleBinding` connecting the global `ClusterRole` with the service account of the operator that will manage N's resources.
+
+### Advantages and constraints
+
+The primary benefit of this architecture is that it enables different tenants to manage their operators independently along with their credentials and resources.
+Additionally in the case of operators using managed identities it removes the need to replicate `AzureIdentity` and `AzureIdentityBinding` resources between namespaces to ensure that the pod identity machinery configures the node identities correctly for the operator.
+
+The biggest downside (other than complexity) is that the different operator installations will be separately managed and we can't assume that they will all be upgraded at the same time.
+The CRDs and webhook service deployment will need to be upgraded by the cluster administrator before any of the tenant operators can be upgraded.
+Conversely if there's a release of the operator that removes an old version of a resource definition, the cluster administrator will need to ensure that all of the tenant operators and resources have been upgraded beyond that version before the core CRDs and webhook service are upgraded.
```

Comment:
A version compatibility matrix between the cluster admin components (CRD and webhook) and tenant components (namespaced operators) would be useful for validating this.